hconfig_utils/hconfig_get_private.F90

```diff
@@ -146,4 +164,104 @@ function make_fmt(descriptor) result(fmt)
 
    end function make_fmt
 
+#include "mapl_hconfig_macros.h"
+
+#define _SUB_ mapl_get_value_I4
+#define _FTYPE_ integer(kind=ESMF_KIND_I4)
+#define _ESMF_FUNC_ ESMF_HConfigAsI4
+#define _TYPESTRING_ "I4"
+#define _EDIT_DESCRIPTOR_ "G0"
+#include "mapl_hconfig_get_value_template.h"
+#include "mapl_hconfig_get_value_template_undef.h"
```

Comment:
Don't you want/need some #undef's after each block here?    There is some danger of definitions leaking through and would be very hard to debug.  Esp. concerned about _IS_LOGICAL_ leaking down.

Reply:
`_IS_LOGICAL_` is undefined in `mapl_hconfig_get_value_template_undef.h`. I kept the logical calls at the end just in case.